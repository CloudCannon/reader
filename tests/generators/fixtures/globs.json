{
	"source": "tests/generators/fixtures/standard",
	"data-config": {
		"authors": {
			"path": "data/authors.csv"
		},
		"fruit": {
			"path": "data/fruit"
		}
	},
	"collections-config": {
		"posts": {
			"path": "_posts",
			"glob": [
				"tests/generators/fixtures/standard/_posts/2016-08-12-*",
				"**/nested/*"
			],
			"output": true,
			"url": "hello/{title|slugify}"
		},
		"vegetables": {
			"path": "vegetables",
			"glob": "**/po*"
		}
	},
	"_comments": {},
	"_options": {},
	"_array_structures": {},
	"_select_data": {},
	"generator": {},
	"_source_editor": {},
	"paths": {
		"uploads": "assets/uploads"
	},
	"base-url": ""
}
