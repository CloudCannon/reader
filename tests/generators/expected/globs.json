{
  "source": "tests/generators/fixtures/standard",
  "data-config": {
    "authors": {
      "path": "data/authors.csv"
    },
    "fruit": {
      "path": "data/fruit"
    }
  },
  "collections-config": {
    "posts": {
      "path": "_posts",
      "glob": [
        "tests/generators/fixtures/standard/_posts/2016-08-12-*",
        "**/nested/*"
      ],
      "output": true,
      "url": "hello/{title|slugify}"
    },
    "vegetables": {
      "path": "vegetables",
      "glob": "**/po*"
    }
  },
  "_comments": {},
  "_options": {},
  "_array_structures": {},
  "_select_data": {},
  "generator": {},
  "_source_editor": {},
  "paths": {
    "uploads": "assets/uploads"
  },
  "base-url": "",
  "time": "2000-11-22T00:00:00.000Z",
  "cloudcannon": {
    "name": "cloudcannon-reader",
    "version": "0.0.1"
  },
  "data": {
    "authors": [
      {
        "name": "Jim",
        "interest": "kayaks"
      },
      {
        "name": "Pam",
        "interest": "art"
      }
    ],
    "fruit": [
      {
        "name": "Apple",
        "rating": 5
      },
      {
        "name": "Banana",
        "rating": 4
      }
    ]
  },
  "collections": {
    "posts": [
      {
        "date": "2020-08-12T00:00:00.000Z",
        "title": "The history of marketing",
        "categories": [
          "marketing"
        ],
        "author_staff_member": "robin",
        "path": "_posts/2016-08-12-the-history-of-marketing.md",
        "collection": "posts",
        "url": "hello/the-history-of-marketing"
      },
      {
        "date": "2020-07-22T00:00:00.000Z",
        "title": "Egg",
        "categories": [
          "tips"
        ],
        "author_staff_member": "betty",
        "path": "_posts/nested/2020-07-22-egg.md",
        "collection": "posts",
        "url": "hello/egg"
      }
    ],
    "vegetables": [
      {
        "name": "Potato",
        "rating": 5,
        "path": "vegetables/potato.yml",
        "collection": "vegetables",
        "url": ""
      }
    ]
  }
}
